src/OpenLoco/Graphics/PaletteMap.cpp

```diff
@@ -0,0 +1,151 @@
+#include "PaletteMap.h"
+#include "../Drawing/SoftwareDrawingEngine.h"
+#include "../Interop/Interop.hpp"
+#include "Gfx.h"
+#include "ImageIds.h"
+#include <algorithm>
+#include <cassert>
+#include <numeric>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::Gfx
+{
+    static loco_global<uint32_t[147], 0x050B8C8> _paletteToG1Offset;
+    static loco_global<Drawing::PaletteEntry[256], 0x0113ED20> _113ED20;
+
+    const PaletteMap& PaletteMap::getDefault()
+    {
+        static bool initialised = false;
+        static uint8_t data[256];
+        static PaletteMap defaultMap(data);
+        if (!initialised)
+        {
+            std::iota(std::begin(data), std::end(data), 0);
```

Comment:
It was pointed out in openrct2 that we never set the initialised variable to true.

Reply:
I just moved things around and didn't really look for the details, but I can probably improve that part.